src/Compiler/Service/IncrementalBuild.fs

```diff
@@ -138,11 +139,11 @@ module IncrementalBuildSyntaxTree =
                 )
             ), sourceRange, fileName, [||]
 
-        let parse _ =
-            let diagnosticsLogger = CompilationDiagnosticLogger("Parse", tcConfig.diagnosticsOptions)
-            // Return the disposable object that cleans up
-            use _holder = new CompilationGlobalsScope(diagnosticsLogger, BuildPhase.Parse)
-            try           
+        let getParseTask (source: FSharpSource) =
+            backgroundTask {
```

Comment:
I might have misunderstood how backgroundTask works. The intent was to start that job on thread pool as soon as possible. Async.StartAsTask?

Reply:
If you want it on the threadpool, you probably want to `Task.Run` it.

But i'm slightly confused here too.

Do I understand the intention correctly - you want it to start (on background) as soon as it's created, so when you need an actual result, it would be hightly probable that it's already started (or even finished)?

Reply:
Then, yes, you probably want to have it running via `Task.Run`.

Reply:
thanks @vzarytovskii, it does help, with Task.Run I can see the gains in FileCascade benchmark.